### ISSUE TITLE
Site creation fails because of conflicts with obsolete bench cleanups

### ISSUE THREAD

**Issue by adityahase** [frappe/press#1845]
It's possible to have a site creation fail because the bench underneath vanishes during unused bench cleanups.

Here's a sample case

|Index|Job|Timestamp|State|Reference field|
|----|----|----|----|----|
|1|Archive Bench|2024-06-04 19:18:49|Job created|creation|
|2|Archive Bench|04-06-2024 19:18:51|Job started|start (UTC)|
|3|Archive Bench|2024-06-04 19:18:49|Job completed|end (UTC)|
|4|Archive Bench|2024-06-04 19:18:58|Job status updated |modified|
|5|Archive Bench|2024-06-04 19:19:05|Bench status updated|Bench.modified|
|6|New Site|2024-06-04 19:19:53|Site created|Site.creation|
|7|New Site|2024-06-04 19:19:55|Job created|creation|

The New Site checks the Bench status (Active). Since step 6 happened after ~50 seconds after step 5. It should've been already set to Archive at the time.

Found trace id `FRAPPE_TRACE_ID: 75a6b22a-665f-4909-82da-cd8d943c8b9a ` with `mysqlbinlog mysql-bin.002017 | grep -A5 -B5 "0ee16af459" | less -SFX`

And monitor.json.log shows 
```JSON
{
  "duration": 2795552,
  "request": {
    "ip": "xxx.xxx.xxx.xxx",
    "method": "POST",
    "path": "/api/method/press.api.client.insert",
    "status_code": 500
  },
  "site": "frappe.cloud",
  "timestamp": "2024-06-04 13:49:53.584219+00:00",
  "transaction_type": "request",
  "uuid": "75a6b22a-665f-4909-82da-cd8d943c8b9a"
}
```
Not sure about status_code 500. NGINX shows 200.

```nginx
xxx.xxx.xxx.xxx - - [04/Jun/2024:13:49:56 +0000] "POST /api/method/press.api.client.insert HTTP/2.0" 200 1117 "https://frappecloud.com/dashboard/benches/bench-xxxxx/sites/new" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36" "-" "frappecloud.com" 2.805 ct=0.000 rt=
```

The bench was archived from this job

```JSON
{
  "duration": 274750,
  "job": {
    "method": "press.press.doctype.bench.bench.archive_obsolete_benches_for_server",
    "scheduled": false,
    "wait": 1891291
  },
  "site": "frappe.cloud",
  "timestamp": "2024-06-04 13:48:49.444857+00:00",
  "transaction_type": "job",
  "uuid": "frappe.cloud::archive_obsolete_benches:fxxx-mumbai.frappe.cloud"
}
```
FYI Archive Bench checks for 
- No pending/running jobs
- No sites

References:
- Archive Bench job https://frappecloud.com/app/agent-job/06790fd099
- New Site job https://frappecloud.com/app/agent-job/0ee16af459

---
I don't see any conflicts here. This shouldn't have happened.

---
#### Update:

The UI explicitly passes the bench :man_facepalming: 
https://github.com/frappe/press/blob/dbe6e7cdbc583b551bde8fc7a86dfc8c9f1b1137/dashboard/src2/pages/NewSite.vue#L322

This wouldn't have happened with https://github.com/frappe/press/blob/dbe6e7cdbc583b551bde8fc7a86dfc8c9f1b1137/press/api/site.py#L102-L122

client.py insert ignores these validations/heuristics.
https://github.com/frappe/press/blob/dbe6e7cdbc583b551bde8fc7a86dfc8c9f1b1137/press/press/doctype/site/site.py#L254-L261

Let's use the old validations as well? @netchampfaris @BreadGenie

**Comment by adityahase** [comment]
https://github.com/frappe/press/commit/b6df717543fe1371a8b498f5a384635de8018f96 Shoul help

**Comment by netchampfaris** [comment]
@adityahase 

That query is present in site.py _set_latest_bench method. 

https://github.com/frappe/press/blob/ba26ef4a175a083ee037c6180e9e08d39383ce8a/press/press/doctype/site/site.py#L1708-L1724

Maybe we should always set the latest bench regardless of whether bench is passed or not?

**Comment by balamurali27** [comment]
fixed by #1906
